experimental/vertx/templates/default/src/main/java/org/acme/quickstart/HttpApplication.java

```diff
@@ -0,0 +1,52 @@
+package org.acme.quickstart;
+
+import io.vertx.core.AbstractVerticle;
+import io.vertx.core.Future;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.ext.web.handler.StaticHandler;
+
+import static io.vertx.core.http.HttpHeaders.CONTENT_TYPE;
+
+public class HttpApplication extends AbstractVerticle {
+
+  protected static final String template = "Hello, %s!";
+
+  @Override
+  public void start(Future<Void> future) {
+    // Create a router object.
+    Router router = Router.router(vertx);
+
+    router.get("/api/greeting").handler(this::greeting);
+    router.get("/*").handler(StaticHandler.create());
+
+    // Create the HTTP server and pass the "accept" method to the request handler.
+    vertx
+        .createHttpServer()
+        .requestHandler(router)
+        .listen(
+            // Retrieve the port from the configuration, default to 8080.
+            config().getInteger("http.port", 8080), ar -> {
+              if (ar.succeeded()) {
+                System.out.println("Server started on port " + ar.result().actualPort());
```

Comment:
nit: maybe we can have the full url of where the server has started rather than just the port number?

Reply:
will it not always be `localhost`?

Reply:
It will be, I just thought it might be clearer to see the whole thing.